server/src/main/java/org/opensearch/index/cache/request/RequestCacheStats.java

```diff
@@ -48,69 +52,127 @@
  */
 public class RequestCacheStats implements Writeable, ToXContentFragment {
 
-    private long memorySize;
-    private long evictions;
-    private long hitCount;
-    private long missCount;
+    private Map<String, StatsHolder> map;
 
-    public RequestCacheStats() {}
+    public RequestCacheStats() {
+        this.map = new HashMap<>();
+        for (TierType tierType : TierType.values()) {
+            map.put(tierType.getStringValue(), new StatsHolder());
+            // Every possible tier type must have counters, even if they are disabled. Then the counters report 0
+        }
+    }
 
     public RequestCacheStats(StreamInput in) throws IOException {
-        memorySize = in.readVLong();
-        evictions = in.readVLong();
-        hitCount = in.readVLong();
-        missCount = in.readVLong();
+        this();
```

Comment:
We are calling to initialize this map but looks error prone and wrong logically. As logically all variables inside this constructor should be initialized with StreamInput values. You can initialize the map inline as suggested above and we don't need this.